### PR TITLE
Fix issue #5507 "Constructor WebServer::WebServer(IPAddress addr, int…

### PR DESCRIPTION
**Pull request by timr49**
… port) produces an unexpected result"

"The class Webserver is declared with two explicit constructors, one with signature:
WebServer::WebServer(IPAddress addr, int port)
Using this results in a server listening on the port number obtained by converting the value of the IPAddress addr argument (in host byte order) to a uint32_t and then to a uint16_t, which is manifestly not the result that would be expected.
...
As for a fix, we can assume from these results that this constructor is not being used and therefore could simply be deleted."

----------------------------------------------------------------------------------------------------------------------------------------------------
This entire section can be deleted if all items are checked.

*By completing this PR sufficiently, you help us to improve the quality of Release Notes*

### Checklist

1. [x] Please provide specific title of the PR describing the change, including the component name (eg."Update of Documentation link on Readme.md")
2. [x] Please provide related links (eg. Issue, other Project, submodule PR..)
----------------------------------------------------------------------------------------------------------------------------------------------------

## Summary
Fix issue #5507 "Constructor WebServer::WebServer(IPAddress addr, int port) produces an unexpected result"

## Impact
This change deletes the declaration and invalid definition of an apparently unused constructor of class WebServer. The intended/expected impact is that (attempted) use of this constructor will result in a compile-time error rather than erroneous run-time behaviour.

### CODE REVIEW

**Comment by Rotzbua** [comment]
Wouldn't it be better to implement the missing function like in old esp:
https://github.com/esp8266/Arduino/blob/3.0.2/libraries/ESP8266WiFi/src/WiFiServer.h#L80

Furthermore removing maybe break derived classes.. so better fix `, _server(addr, port)` and print a warning at runtime?

**Comment by me-no-dev** [comment]
I agree with @Rotzbua . Issue should be fixed in `WiFiServer`

**Comment by me-no-dev** [comment]
@timr49 even though there are no examples, in reality we can/should add this functionality, because ESP-IDF supports it and could be useful in some cases. ESP32 can have more than 3 interfaces at the same time :) For now the code can just go to the  other constructor.

**Comment by timr49** [comment]
> @timr49 even though there are no examples, in reality we can/should add this functionality, because ESP-IDF supports it and could be useful in some cases. ESP32 can have more than 3 interfaces at the same time :) For now the code can just go to the other constructor.

It occurred to me after commenting above that the ESP WiFi can have two IP interfaces, one the station and one the soft AP.  So, we do have a way of testing multi-homed servers, at least for n=2.

**Comment by timr49** [comment]
I have pushed a different change onto the same branch, which adds support for multi-homed web servers. It  includes a new examples script (WebServer/MultiHomedServers), which test this change. The testing method is in the comments. The expected results were achieved.

At about the same time as my push, I got some error notifications from CI, but am not sure whether they are related to my previous fix or this one. Unfortunately, my GitHub skills are new, so am not sure what to do next.

**Comment by timr49** [comment]
> At about the same time as my push, I got some error notifications from CI, but am not sure whether they are related to my previous fix or this one. Unfortunately, my GitHub skills are new, so am not sure what to do next.

I successfully guessed what I had done wrong and fixed it in a further commit to the pull request branch ("All checks have passed").

I believe that this change is now ready for review.

File: libraries/WebServer/examples/MultiHomedServers/MultiHomedServers.ino [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  Serial.print(" IP address: ");
+  Serial.println(WiFi.softAPIP());
+
+  if (MDNS.begin("esp32")) {
+    Serial.println("MDNS responder started");
+  }
+
+  server0 = new WebServer(8080);
+  server1 = new WebServer(WiFi.localIP(), 8081);
+  server2 = new WebServer(WiFi.softAPIP(), 8082);
```
Comment (me-no-dev) [inline libraries/WebServer/examples/MultiHomedServers/MultiHomedServers.ino:150]: for this test to be valid, both servers should be on the same port. That is how you can make sure that the functionality works.

Comment (timr49) [reply libraries/WebServer/examples/MultiHomedServers/MultiHomedServers.ino]: Changed port numbers as suggested. Expected results returned by test script. Pushed new commit. "All checks have passed".

**Comment by me-no-dev** [comment]
Thanks @timr49 !!! all merged :)

**Pull request merged**